app/controllers/collection_items_controller.rb

```diff
@@ -130,9 +154,15 @@ def create
       flash[:notice] = ts("Added to collection(s): %{collections}.",
                             :collections => new_collections.collect(&:title).join(", "))
     end
+    unless invited_collections.empty?
+      invited_collections.each do |needs_user_approval|
+        flash[:notice] ||= ""
+        flash[:notice] = ts("This work has been <a href=\"#{collection_items_path(needs_user_approval)}?invited=true\">Invited</a> to your collection (#{needs_user_approval.title}).").html_safe
+      end
+    end
     unless unapproved_collections.empty?
-      flash[:notice] += "<br />" + ts("Your addition will have to be approved before it appears in %{moderated}.",
-        :moderated => unapproved_collections.collect(&:title).join(", "))
+      flash[:notice] ||= ""
+      flash[:notice] += ts(" You have submitted your work to #{unapproved_collections.size > 1 ? "moderated collections (%{all_collections}). It will not become a part of those collections" : "the moderated collection '%{all_collections}'. It will not become a part of the collection"} until it has been approved by a moderator.", all_collections: unapproved_collections.map { |f| f.title }.join(', '))
```

Comment:
Prefer single-quoted strings inside interpolations.<br>Pass `&:title` as an argument to `map` instead of a block.